src/debugger/src/debugger.cpp

```diff
@@ -0,0 +1,27 @@
+/**
+ * @file debugger.cpp
+ * Defines the Debugger class
+ */
+
+#include "debugger/debugger.h"
+#include "gameboy/gameboy.h"
+
+namespace debugger {
+
+Debugger::Debugger(std::unique_ptr<gameboy::Gameboy> gameboy)
+    : gameboy(std::move(gameboy)) {
+
+	// TODO
+	cpu = nullptr;
+	gpu = nullptr;
+	cartridge = nullptr;
+}
+
+void Debugger::tick() {
+	ticks++;
```

Comment:
By this do you mean to increment the Debugger ticks? Is this in any way related to the CPU ticks?

Reply:
Hmm, I guess my language is a bit ambiguous here.

There's a distinction between ticks and cycles that I'm trying to make, where ticks is the number of times that `tick` is called, and CPU Cycles are a completely unrelated number, describing the number of real hardware cycles that should have been run at any particular time.

This is why I made the CpuCycles type, but I guess it's still a bit misleading. I'll try to improve this in coming PRs. Great point, thanks!

Reply:
Okay, I think I kind of get it now.